indico/modules/events/registration/util.py

```diff
@@ -438,8 +439,12 @@ def generate_ticket_qr_code(registration):
         "event_id": unicode(registration.event_new.id),
         "server_url": Config.getInstance().getBaseURL()
     }
-    json_qr_data = json.dumps(qr_data)
-    qr.add_data(json_qr_data)
+    extended_qr = signals.event.designer.generate_ticket_qr_code.send(registration, qr_data=qr_data)
```

Comment:
I would rather call the signal with just the registration as the sender and the `qr_data` kwarg (but maybe renamed to `ticket_data`), and let the signal modify the dict directly instead of returning a new one. Your current code won't work well when there's more than one plugin connecting to the signal.

Generally it's a good idea to use `values_from_signal(...)` when dealing with signals that return values, but if you go for changing the dict directly you won't need that.